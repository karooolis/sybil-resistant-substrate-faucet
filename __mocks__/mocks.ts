import { Session } from "next-auth";

export const mockGithubSession: Session = {
  expires: "1",
  provider: "github",
  providerAccountId: "GITHUB_PROVIDER_ACCOUNT_ID",
  user: {
    email: "user@mail.com",
    name: "USER_NAME",
  },
};

export const mockTwitterSession: Session = {
  expires: "1",
  provider: "twitter",
  providerAccountId: "TWITTER_PROVIDER_ACCOUNT_ID",
  user: {
    name: "USER_NAME",
  },
};
