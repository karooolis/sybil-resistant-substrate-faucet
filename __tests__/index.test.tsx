import { render } from "@testing-library/react";
import { useSession } from "next-auth/react";
import { mockGithubSession, mockTwitterSession } from "../__mocks__/mocks";
import Home from "../pages/index";

jest.mock("next-auth/react", () => {
  return {
    useSession: jest.fn(),
  };
});

describe("index page", () => {
  beforeEach(() => {
    jest.resetModules();
  });

  it("renders homepage unchanged - not logged in", () => {
    (useSession as jest.Mock).mockReturnValue({
      data: null,
      status: "unauthenticaated",
    });

    const { container } = render(<Home claimed={false} />);
    expect(container).toMatchSnapshot();
  });

  it("renders homepage unchanged - logged in with GitHub", () => {
    (useSession as jest.Mock).mockReturnValue({
      data: mockGithubSession,
      status: "authenticaated",
    });

    const { container } = render(<Home claimed={false} />);
    expect(container).toMatchSnapshot();
  });

  it("renders homepage unchanged - logged in with Twitter", () => {
    (useSession as jest.Mock).mockReturnValue({
      data: mockTwitterSession,
      status: "authenticaated",
    });

    const { container } = render(<Home claimed={false} />);
    expect(container).toMatchSnapshot();
  });
});
