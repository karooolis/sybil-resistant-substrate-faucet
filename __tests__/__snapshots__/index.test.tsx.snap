// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index page renders homepage unchanged - logged in with GitHub 1`] = `
<div>
  <div
    class="max-w-7xl h-screen mx-auto px-4 sm:px-6 lg:px-8"
  >
    <main
      class="max-w-3xl mx-auto"
    >
      <h2
        class="text-center pt-12 text-3xl leading-8 font-bold tracking-tight text-gray-900 sm:text-4xl sm:tracking-tight"
      >
        Polkadot Faucet
      </h2>
      <p
        class="mt-2 max-w-3xl mx-auto text-center text-xl text-gray-500"
      >
        Bootstrap your testnet wallet
      </p>
      <div
        class="mt-12 bg-white shadow border border-gray-300 sm:rounded-lg"
      >
        <div
          class="px-4 py-5 sm:p-6"
        >
          <h3
            class="text-lg leading-6 font-medium text-gray-900"
          >
            Request Tokens
          </h3>
          <p
            class="mt-2 text-gray-900"
          >
            To prevent faucet abuse, you must sign in with Twitter or GitHub.
          </p>
          <button
            class="block mt-4 -mb-1 mx-auto text-blue-500 underline"
          >
            Sign out 
            user@mail.com
          </button>
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`index page renders homepage unchanged - logged in with Twitter 1`] = `
<div>
  <div
    class="max-w-7xl h-screen mx-auto px-4 sm:px-6 lg:px-8"
  >
    <main
      class="max-w-3xl mx-auto"
    >
      <h2
        class="text-center pt-12 text-3xl leading-8 font-bold tracking-tight text-gray-900 sm:text-4xl sm:tracking-tight"
      >
        Polkadot Faucet
      </h2>
      <p
        class="mt-2 max-w-3xl mx-auto text-center text-xl text-gray-500"
      >
        Bootstrap your testnet wallet
      </p>
      <div
        class="mt-12 bg-white shadow border border-gray-300 sm:rounded-lg"
      >
        <div
          class="px-4 py-5 sm:p-6"
        >
          <h3
            class="text-lg leading-6 font-medium text-gray-900"
          >
            Request Tokens
          </h3>
          <p
            class="mt-2 text-gray-900"
          >
            To prevent faucet abuse, you must sign in with Twitter or GitHub.
          </p>
          <button
            class="block mt-4 -mb-1 mx-auto text-blue-500 underline"
          >
            Sign out 
            USER_NAME
          </button>
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`index page renders homepage unchanged - not logged in 1`] = `
<div>
  <div
    class="max-w-7xl h-screen mx-auto px-4 sm:px-6 lg:px-8"
  >
    <main
      class="max-w-3xl mx-auto"
    >
      <h2
        class="text-center pt-12 text-3xl leading-8 font-bold tracking-tight text-gray-900 sm:text-4xl sm:tracking-tight"
      >
        Polkadot Faucet
      </h2>
      <p
        class="mt-2 max-w-3xl mx-auto text-center text-xl text-gray-500"
      >
        Bootstrap your testnet wallet
      </p>
      <div
        class="mt-12 bg-white shadow border border-gray-300 sm:rounded-lg"
      >
        <div
          class="px-4 py-5 sm:p-6"
        >
          <h3
            class="text-lg leading-6 font-medium text-gray-900"
          >
            Request Tokens
          </h3>
          <p
            class="mt-2 text-gray-900"
          >
            To prevent faucet abuse, you must sign in with Twitter or GitHub.
          </p>
          <div
            class="mt-2"
          >
            <button
              class="mt-3 w-full inline-flex items-center justify-center px-4 py-2 border border-transparent shadow-sm font-medium rounded-md text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
              type="button"
            >
              Sign in with Twitter
            </button>
            <button
              class="mt-3 w-full inline-flex items-center justify-center px-4 py-2 border border-transparent shadow-sm font-medium rounded-md text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
              type="button"
            >
              Sign in with GitHub
            </button>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;
